### PR TITLE
Fix: Redirect to the login page when accessing private ontology while being not authenticated.

### PR DESCRIPTION
**Pull request by Bilelkihal**
### Related issue
https://github.com/agroportal/project-management/issues/563

### Changes
- Redirect the user to the login page when trying to access a private ontology while being not logged in.
- Use alert component for login errors to display them better.
- Add the option of notices to login URL, so we can write notices when we redirect to login page.
- Add a message in the login page to say, this is a private ontology please login when trying to access a private ontology while being not logged in.

### Demo
https://github.com/ontoportal-lirmm/bioportal_web_ui/assets/61744974/585744d4-09e1-4d5d-a59e-b1882df769a6

### CODE REVIEW

**Comment by Bilelkihal** [comment]
@syphax-bouazzouni I started this branch from the stage branch (so I can launch the dev env) you can see that the commits history is divergent than the one of of the development, to review this PR, just see the last three commits or these five files:
`app/controllers/ontologies_controller.rb`
`app/views/login/index.html.haml`
`app/controllers/login_controller.rb`
`config/locales/en.yml`
`config/locales/fr.yml`

**Review by Bilelkihal — COMMENTED** [review]
do a merge for the development branch here once the changes are approved

**Comment by jonquet** [comment]
Top! perfect.

File: app/helpers/check_resolvability_helper.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def resolvability_formats
   end
 
   def resolvability_timeout
-    5
+    5000
```
Comment (syphax-bouazzouni) [inline app/helpers/check_resolvability_helper.rb:22]: remove this change

File: Gemfile.lock [hunk 2]
```diff
@@ -552,7 +552,6 @@ GEM
 PLATFORMS
   x86_64-darwin-23
   x86_64-linux
-  x86_64-linux-gnu
```
Comment (syphax-bouazzouni) [inline Gemfile.lock:555]: remove this change

File: app/views/mappings/mappings_ontologies_table.html.haml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+= render TurboFrameComponent.new(id: 'mappings_table') do
```
Comment (syphax-bouazzouni) [inline app/views/mappings/mappings_ontologies_table.html.haml:1]: this file is to remove

File: app/controllers/ontologies_controller.rb [hunk 4]
```diff
@@ -241,7 +241,15 @@ def show
 
     # Note: find_by_acronym includes ontology views
     @ontology = LinkedData::Client::Models::Ontology.find_by_acronym(params[:ontology]).first
-    ontology_not_found(params[:ontology]) if @ontology.nil? || @ontology.errors
+    
+    if @ontology.nil? || @ontology.errors
+      if @ontology&.errors.first.eql?("Access denied for this resource")
+        redirect_to "/login?redirect=/ontologies/#{params[:ontology]}&notices=#{escape(t('login.private_ontology'))}"
```
Comment (syphax-bouazzouni) [inline app/controllers/ontologies_controller.rb:247]:
```ruby 
redirect_to "/login?redirect=/ontologies/#{params[:ontology]}&notices=#{escape(t('login.private_ontology'))}", notice: "yourt message"
```

File: app/controllers/login_controller.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@ class LoginController < ApplicationController
   layout :determine_layout
 
   def index
+    @notices = [params[:notices]] if params[:notices]
```
Comment (syphax-bouazzouni) [inline app/controllers/login_controller.rb:6]: not needed if your use rails notices

File: app/views/instances/_details.html.haml [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
       =  render ConceptDetailsComponent.new(id:'instance-details', acronym: ontology_acronym, concept_id: @instance["@id"]) do |c|
         - c.header(stripped: true) do |t|
-          - t.add_row({th: t("instances.id")}, {td: link_to_with_actions(@instance["@id"]) })
+          - t.add_row({th: t("collections.id")}, {td: link_to_with_actions(@instance["@id"], acronym: @ontology.acronym) })
```
Comment (syphax-bouazzouni) [inline app/views/instances/_details.html.haml:9]: change "collections.id" with "instances.id"

File: config/routes.rb [hunk 7]
```diff
@@ -32,13 +32,15 @@
 
   post '/mappings/loader', to: 'mappings#loader_process'
   get 'mappings/count/:id', to: 'mappings#count', constraints: { id: /.+/ }
+  get 'mappings/ontology_mappings/:acronym', to: 'mappings#ontology_mappings'
```
Comment (syphax-bouazzouni) [inline config/routes.rb:35]: to remove

File: config/routes.rb [hunk 8]
```diff
@@ -32,13 +32,15 @@
...
   get 'mappings/show_mappings', to: 'mappings#show_mappings'
   get 'mappings/new', to: 'mappings#new'
   get 'mappings/:id', to: 'mappings#show', constraints: { id: /.+/ }
   post 'mappings/:id', to: 'mappings#update', constraints: { id: /.+/ }
   delete 'mappings/:id', to: 'mappings#destroy', constraints: { id: /.+/ }
   resources :mappings
   get 'mappings/:id', to: 'mappings#show', constraints: { id: /.+/ }
+  get '/mappings_ontologies_table', to: 'mappings#mappings_ontologies_table'
```
Comment (syphax-bouazzouni) [inline config/routes.rb:43]: to remove

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Review by Bilelkihal — COMMENTED** [review]
✅

**Pull request merged**